src/builder.rs

```diff
@@ -6,18 +6,32 @@ use utils::Toggle;
 use params::NoiseParams;
 use resolvers::CryptoResolver;
 use error::{SnowError, InitStage, Prerequisite};
+use subtle::ConstantTimeEq;
 
 /// A keypair object returned by [`generate_keypair()`]
 ///
 /// [`generate_keypair()`]: #method.generate_keypair
-#[derive(PartialEq)]
 pub struct Keypair {
     /// The private asymmetric key
     pub private: Vec<u8>,
     /// The public asymmetric key
     pub public: Vec<u8>,
 }
 
+
+impl PartialEq for Keypair {
+    fn eq(&self, other: &Keypair) -> bool {
+        let priv_eq = self.private.ct_eq(&other.private).unwrap_u8() == 1u8;
+        let pub_eq = self.public.ct_eq(&other.public).unwrap_u8() == 1u8;
+
+        if priv_eq & pub_eq {
```

Comment:
actually, looking at the documentation for `subtle`, I see their `Choice` implements `BitAnd<Choice>` and `From<Choice>` for `bool`, which would all be preferable to us implementing our own conversions.

I haven't tested this, but wouldn't this work?

```rust
let priv_eq = self.private.ct_eq(&other.private);
let pub_eq = self.public.ct_eq(&other.public);
(priv_eq & pub_eq).into()
```

Reply:
Good idea.